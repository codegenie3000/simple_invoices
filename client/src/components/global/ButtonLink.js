import React from 'react';
import globalStyles from '../../stylesheets/GlobalElements.module.css';
import {withRouter} from 'react-router-dom';

const buttonMargin = {
    margin: '10px auto auto 0'
};

function ButtonPrimary (props) {
    const {to, label, history, onClick, location, match, staticContext, ...rest} = props;
    return (
        <button
            {...rest}
            className={globalStyles.buttonPrimary}
            onClick={event => {
                onClick && onClick(event);
                history.push(to)
            }}
            >{label}
        </button>
    );
}

export default withRouter(ButtonPrimary);
