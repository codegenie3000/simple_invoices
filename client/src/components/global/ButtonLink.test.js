import React from 'react';
import { render, mount, shallow } from 'enzyme';
import { BrowserRouter, withRouter, Route } from 'react-router-dom';

import ButtonLink from './ButtonLink';

describe('Button Link', () => {
    it('should render correctly', () => {
        const component = render(withRouter(
            <ButtonLink
                to={ '/foo' }
                label={ 'foo' }
                history={ 'foo' }

            />));
        expect(component).toMatchSnapshot();
    });
    it('should receive a click', () => {
        const component = mount(
            <BrowserRouter>
                <Route>
                    <ButtonLink
                        to={ '/foo' }
                        label={ 'foo' }
                        history={ 'foo' }
                        onClick={ () => {
                        } }
                    />
                </Route>
            </BrowserRouter>
        );
        component
            .find('ButtonLink')
            .simulate('click');
        expect(component).toMatchSnapshot();
        component.unmount();
    });
});