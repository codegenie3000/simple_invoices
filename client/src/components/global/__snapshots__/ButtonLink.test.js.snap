// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Link should receive a click 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "PUSH",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 2,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "6m0xme",
          "pathname": "/foo",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Route>
      <withRouter(ButtonLink)
        history="foo"
        label="foo"
        onClick={[Function]}
        to="/foo"
      >
        <Route>
          <ButtonLink
            history={
              Object {
                "action": "PUSH",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 2,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "key": "6m0xme",
                  "pathname": "/foo",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            label="foo"
            location={
              Object {
                "hash": "",
                "key": "6m0xme",
                "pathname": "/foo",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": false,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            onClick={[Function]}
            to="/foo"
          >
            <button
              className="buttonPrimary"
              onClick={[Function]}
            >
              foo
            </button>
          </ButtonLink>
        </Route>
      </withRouter(ButtonLink)>
    </Route>
  </Router>
</BrowserRouter>
`;

exports[`Button Link should render correctly 1`] = `null`;
